In [1]:
# import
import numpy as np;
import matplotlib.pyplot as plt;
import pandas as pd

In [4]:
# convert the excel file to csv file via online tool.
df = pd.read_csv("marital status.csv")

In [5]:
df # shows the file data

age  status
0    21     0.0
1    33     1.0
2    24     0.0
3    28     1.0
4    17     0.0
5    33     1.0
6    23     0.0
7    30     0.0
8    29     1.0
9    22     0.0
10   26     1.0
11   20     0.0
12   22     0.0
13   24     NaN
14   34     1.0
15   29     1.0
16   16     0.0
17   18     0.0
18   26     1.0
19   25     1.0

In [8]:
df.isnull().sum()

age       0
status    1
dtype: int64

In [9]:
handleStatusNAN = df['status'].median()

In [10]:
handleStatusNAN

0.0

In [11]:
df.status = df.status.fillna(handleStatusNAN)

In [12]:
df

age  status
0    21     0.0
1    33     1.0
2    24     0.0
3    28     1.0
4    17     0.0
5    33     1.0
6    23     0.0
7    30     0.0
8    29     1.0
9    22     0.0
10   26     1.0
11   20     0.0
12   22     0.0
13   24     0.0
14   34     1.0
15   29     1.0
16   16     0.0
17   18     0.0
18   26     1.0
19   25     1.0

In [13]:
df['status'].value_counts()

0.0    11
1.0     9
Name: status, dtype: int64

In [14]:
# feature sepearting
x = df[['age']]

In [15]:
x

age
0    21
1    33
2    24
3    28
4    17
5    33
6    23
7    30
8    29
9    22
10   26
11   20
12   22
13   24
14   34
15   29
16   16
17   18
18   26
19   25

In [16]:
y = df['status']

In [17]:
y

0     0.0
1     1.0
2     0.0
3     1.0
4     0.0
5     1.0
6     0.0
7     0.0
8     1.0
9     0.0
10    1.0
11    0.0
12    0.0
13    0.0
14    1.0
15    1.0
16    0.0
17    0.0
18    1.0
19    1.0
Name: status, dtype: float64

In [18]:
from sklearn.model_selection import train_test_split

In [19]:
# spliting training and test data set. 0.3 means 30% of the data set and random state is randomly taken variable
xtrain,xtest,ytrain,ytest = train_test_split(x,y, test_size = 0.3, random_state =1)

In [20]:
from sklearn.linear_model import LogisticRegression

In [21]:
model = LogisticRegression()

In [22]:
model

LogisticRegression()

In [23]:
model.fit(xtrain,ytrain)

LogisticRegression()

In [24]:
# predic the test data set
model.predict(xtest)

array([1., 0., 0., 0., 0., 1.])

In [25]:
# check accuracy
model.score(xtest,ytest)

0.8333333333333334

In [29]:
# logistic regression sigmoid function value check
model.predict_proba(xtest)

array([[0.28697051, 0.71302949],
       [0.9902967 , 0.0097033 ],
       [0.80161052, 0.19838948],
       [0.50311753, 0.49688247],
       [0.71810555, 0.28189445],
       [0.02465085, 0.97534915]])

### Observation : The first value in train the probability , the first part is for no and second part is fro yes . So from the above values the fist value's no is 28% where yes is 71%. so the answer is 1 